File: backend/Boat.java
public class Boat {
    private final int id;
    private int size;
    private int partsLeft;
    private int row;
    private int col;

    public Boat(int id) {
        this.id = id;

        switch (id) {
            case 1 -> size = 2;
            case 2, 3 -> size = 3;
            case 4 -> size = 4;
            case 5 -> size = 5;
        }

        partsLeft = size;
    }

    public int getSize() {
        return size;
    }

    public int getPartsLeft() {
        return partsLeft;
    }

    public boolean hit() {
        if (partsLeft > 0) {
            partsLeft--;
            return true;
        }
        return false;
    }

    public int getRow() {
        return row;
    }

    public int getCol() {
        return col;
    }

    public void setRow(int row) {
        this.row = row;
    }

    public void setCol(int col) {
        this.col = col;
    }

    public int getId() {
        return id;
    }
}

File: frontend/src/classes/boat.ts
export class Boat {

    id: number;
    size: number;
    partsLeft: number;

    constructor(id: number) {
        this.id = id;

        switch(id) {
            case 1:
                this.size = 2;
            case 2:
                this.size = 3;
            case 3:
                this.size = 3;
            case 4:
                this.size = 4;
            case 5:
                this.size = 5;
        }

        this.partsLeft = this.size;
    }

    removePart() {
        this.partsLeft--;
    }
}

